tests/integration_tests/tests/connect_info.rs

```diff
@@ -48,3 +48,80 @@ async fn getting_connect_info() {
 
     jh.await.unwrap();
 }
+
+#[cfg(unix)]
+pub mod unix {
+    use std::{convert::TryFrom as _, path::Path};
+
+    use futures_util::FutureExt;
+    use tokio::{
+        net::{UnixListener, UnixStream},
+        sync::oneshot,
+    };
+    use tokio_stream::wrappers::UnixListenerStream;
+    use tonic::{
+        transport::{server::UdsConnectInfo, Endpoint, Server, Uri},
+        Request, Response, Status,
+    };
+    use tower::service_fn;
+
+    use integration_tests::pb::{test_client, test_server, Input, Output};
+
+    struct Svc {}
+
+    #[tonic::async_trait]
+    impl test_server::Test for Svc {
+        async fn unary_call(&self, req: Request<Input>) -> Result<Response<Output>, Status> {
+            let conn_info = req.extensions().get::<UdsConnectInfo>().unwrap();
+
+            // Client-side unix sockets are unnamed.
+            assert!(req.remote_addr().is_none());
+            assert!(conn_info.peer_addr.as_ref().unwrap().is_unnamed());
+            // This should contain process credentials for the client socket.
+            assert!(conn_info.peer_cred.as_ref().is_some());
+
+            Ok(Response::new(Output {}))
+        }
+    }
+
+    #[tokio::test]
+    async fn getting_connect_info() {
+        let unix_socket_path = "/tmp/tonic/uds-integration-test";
+        tokio::fs::create_dir_all(Path::new(unix_socket_path).parent().unwrap())
+            .await
+            .unwrap();
```

Comment:
Would it make sense to use https://doc.rust-lang.org/stable/std/env/fn.temp_dir.html and not have to create a dir then we can remove the need to add the `fs` stuff.

Reply:
Sounds reasonable. Only small issue there is that tokio `UnixListener` needs the path to the unix socket to exist on instantiation, but _does not clean it up_ when dropped. So, we need to remove the file at the end of the test or else we won't be able to run the integration tests multiple times (subsequent runs will fail with `AddrInUse`).

Seems like you want to remove the added `tokio::fs` dep, so I should be able to use `std::env::temp_dir` and just do a blocking rm at the end of the test `std::fs::remove_file` if that sounds good to you?

Reply:
An option would be use [pair ](https://docs.rs/tokio/latest/tokio/net/struct.UnixStream.html#method.pair) to avoid the filesystem altogether. You'd have to convert one of them to a `UnixListener` though, maybe through a raw fd.